readthedocs/urls.py

```diff
@@ -67,22 +74,24 @@
     url(r'^accounts/gold/', include('readthedocs.gold.urls')),
 )
 
-if not getattr(settings, 'USE_SUBDOMAIN', False):
-    urlpatterns += docs_urls
+debug_urls = add(
+    patterns(
+        '',  # base view, flake8 complains if it is on the previous line.
+        url('style-catalog/$',
+            TemplateView.as_view(template_name='style_catalog.html')),
+    ),
+    static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+)
 
-urlpatterns += rtd_urls
-urlpatterns += api_urls
-urlpatterns += core_urls
-urlpatterns += i18n_urls
-urlpatterns += money_urls
-urlpatterns += deprecated_urls
+# Export URLs
+groups = [basic_urls, rtd_urls, project_urls, api_urls, core_urls, i18n_urls,
+          money_urls, deprecated_urls]
 
+if not getattr(settings, 'USE_SUBDOMAIN', False):
+    groups.insert(0, docs_urls)
 if getattr(settings, 'ALLOW_ADMIN', True):
-    urlpatterns += admin_urls
+    groups.append(admin_urls)
+if getattr(settings, 'DEBUG', False):
+    groups.append(debug_urls)
 
-if settings.DEBUG:
-    urlpatterns += patterns(
-        '',  # base view, flake8 complains if it is on the previous line.
-        url('style-catalog/$',
-            TemplateView.as_view(template_name='style_catalog.html')),
-    ) + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
+urlpatterns = reduce(add, groups)
```

Comment:
Generally find functional concepts harder to grok when the surrounding code doesn't use them. Not a huge deal in this case, but just my general feeling around folks comfort level w/ this kind of stuff.